test-app/tests/integration/components/bs-collapse-test.ts

```diff
@@ -1,13 +1,26 @@
 import { module } from 'qunit';
 import { setupRenderingTest } from 'ember-qunit';
-import { render, settled, waitFor } from '@ember/test-helpers';
-import hbs from 'htmlbars-inline-precompile';
+import {
+  render,
+  settled,
+  waitFor,
+  type TestContext,
+} from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
 import { delay, test, visibilityClass } from '../../helpers/bootstrap';
 import setupNoDeprecations from '../../helpers/setup-no-deprecations';
 import a11yAudit from 'ember-a11y-testing/test-support/audit';
 import { skipTransition } from 'ember-bootstrap/utils/transition-end';
 import sinon from 'sinon';
 
+interface Context extends TestContext {
```

Comment:
During my button PR I was actually thinking if we should introduce gts for the test-app already? It would eliminate the need for all that context ceremony just to please TS, as templates can directly access variables in scope, so you don't have to assign to `this`. 🤔

Reply:
That might indeed be a good place to start including GTS. I don't think the setup would change that much.
Hopefully it will also allow for a bit neater template formatting. I find that template-lint can be a tad aggressive at times.

Reply:
I'm open to introduce gts. Especially for the tests. But I think we shouldn't couple/block this PR by it. I'm open to refactor of we land gts before merging this one though. At the end of the day it's just a question of we refactor to `<template>` tag before or after merging this PR.

Refactoring to gts has only one trade-off, which I'm aware of: We would loose test coverage for template registry as all components are imported explicitly. I'm happy to accept that trade-off. Especially as we gain test coverage for explicit imports at the same time.

Reply:
> But I think we shouldn't couple/block this PR by it

I didn't mean to say that. Just came to my mind again after reviewing...